static/app/views/dashboardsV2/gridLayout/dashboard.tsx

```diff
@@ -304,3 +335,30 @@ function getDefaultPosition(index: number, displayType: DisplayType) {
     minH: displayType === DisplayType.BIG_NUMBER ? 1 : 2,
   };
 }
+
+function getMobileLayout(desktopLayout, widgets) {
+  if (desktopLayout.length === 0) {
+    // Initial case where the user has no layout saved, but
+    // dashboard has widgets
+    return [];
+  }
+
+  // Filter out layouts to only those that haven't been deleted
+  const expectedGridKeys = new Set(widgets.map(constructGridItemKey));
+  const survivingLayouts = desktopLayout.filter(({i}) => expectedGridKeys.has(i));
+
+  const layoutWidgetPairs = zip(survivingLayouts, widgets);
+
+  // Sort by y and then subsort by x
+  const sorted = sortBy(layoutWidgetPairs, ['0.y', '0.x']) as [Layout, Widget][];
+
+  const mobileLayout = sorted.map(([layout, widget], index) => ({
+    ...layout,
+    x: 0,
+    y: index * 2,
+    w: 2,
+    h: widget.displayType === DisplayType.BIG_NUMBER ? 1 : 2,
+  }));
+
+  return mobileLayout;
+}
```

Comment:
> Should we adjust this to include calculating BIG_NUMBER widgets? In usage, this doesn't seem to break anything in the ui though.

Yeah, it doesn't seem to break anything because RGL will do vertical compaction so the extra whitespace goes away. That compaction does trigger a re-render but I figured it could be acceptable since this only happens when the user flips between desktop and mobile views, something I feel like is not a common use-case.

> Also assuming desktopLayout is always in sorted order, would we be able to do something simple like this instead?

The sort order is determined by creation and only really used to render the initial view if a person is viewing the dashboard for the first time. After you move things around the ordering stays the same but instead each item gets associated with an `x` and `y` coord that we need to acknowledge if we don't want the ordering to be the "default"